source/benchmark_data.cc

```diff
@@ -479,37 +431,40 @@ void MIT<dim>::compute_wall_data()
           }
 
   // Gather the values of each processor
-  left_boundary_integral   = Utilities::MPI::sum(left_boundary_integral,
-                                                   mpi_communicator);
-  right_boundary_integral  = Utilities::MPI::sum(right_boundary_integral,
-                                                  mpi_communicator);
+  const Triangulation<dim>  &tria = temperature.get_triangulation();
+  const parallel::TriangulationBase<dim> *tria_ptr =
+      dynamic_cast<const parallel::TriangulationBase<dim> *>(&tria);
+  if (tria_ptr != nullptr)
+  {
+    left_boundary_integral   = Utilities::MPI::sum(left_boundary_integral,
+                                                   tria_ptr->get_communicator());
+    right_boundary_integral  = Utilities::MPI::sum(right_boundary_integral,
+                                                   tria_ptr->get_communicator());
+  }
 
-  //Compute and store the Nusselt numbers of the walls
+  // Compute and store the Nusselt numbers of the walls
   nusselt_numbers = std::make_pair(left_boundary_integral/height,
                                    right_boundary_integral/height);
 }
 
 template <int dim>
-void MIT<dim>::compute_global_data()
+void MIT<dim>::compute_global_data(const Entities::FE_VectorField<dim>  &velocity)
 {
-  TimerOutput::Scope  t(*computing_timer, "MIT Benchmark: Global data");
-
-  // Defining the type to contain the vorticity during assembly
+   // Defining the type to contain the vorticity during assembly
   using CurlType = typename FEValuesViews::Vector< dim >::curl_type;
 
   // Initiate the average velocity and vorticity metrics
   average_velocity_metric   = 0.0;
   average_vorticity_metric  = 0.0;
 
   // Quadrature formula
-  const QGauss<dim>   quadrature_formula(velocity->fe_degree() + 1);
+  const QGauss<dim>   quadrature_formula(velocity.fe_degree() + 1);
 
   // Finite element values
-  FEValues<dim> fe_values(*mapping,
-                          velocity->get_finite_element(),
+  FEValues<dim> fe_values(velocity.get_finite_element(),
                           quadrature_formula,
-                          update_values |
-                          update_gradients |
+                          update_values|
+                          update_gradients|
                           update_JxW_values);
```

Comment:
No mapping?

Reply:
I completely removed the mapping from the MIT benchmark since it has only straight boundaries.